components/ILIAS/Mail/classes/class.ilMail.php

```diff
@@ -385,17 +385,68 @@ public function deleteMails(array $mail_ids): void
         }
     }
 
+    private function fetchMailDataFromSaved(?array $row): ?array
+    {
+        if (!is_array($row) || empty($row)) {
+            return null;
+        }
+
+        if (isset($row['attachments'])) {
+            $row['attachments'] = (string) $row['attachments'] ?? NULL;
+            $id = new \ILIAS\ResourceStorage\Identification\ResourceCollectionIdentification($row['attachments']);
+            $row['attachments'] = $id;
+        } else {
+            $row['attachments'] = NULL;
+        }
```

Comment:
Can't we change this to:

```suggestion
        if (isset($row['attachments']) && is_string($row['attachments']) && $row['attachments'] !== '') {
            $row['attachments'] = new ResourceCollectionIdentification($row['attachments']);
        } else {
            $row['attachments'] = null;
        }
```